src/main/java/seedu/storemando/logic/commands/ClearCommand.java

```diff
@@ -1,23 +1,59 @@
 package seedu.storemando.logic.commands;
 
 import static java.util.Objects.requireNonNull;
+import static seedu.storemando.logic.parser.CliSyntax.PREFIX_LOCATION;
+import static seedu.storemando.model.Model.PREDICATE_SHOW_ALL_ITEMS;
 
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.storemando.logic.commands.exceptions.CommandException;
 import seedu.storemando.model.Model;
-import seedu.storemando.model.StoreMando;
+import seedu.storemando.model.item.Item;
+import seedu.storemando.model.item.LocationContainsPredicate;
 
 /**
- * Clears the storemando.
+ * Clears the storemando or a specified location.
  */
 public class ClearCommand extends Command {
 
     public static final String COMMAND_WORD = "clear";
-    public static final String MESSAGE_SUCCESS = "Inventory has been cleared!";
+    public static final String MESSAGE_SUCCESS = "All items in the specified location (if any) are cleared!";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Clear items in storemando or a specified location. "
+        + "Parameters: "
+        + "[" + PREFIX_LOCATION + "LOCATION]\n"
+        + "Example:\n"
+        + "1. " + COMMAND_WORD + "\n"
+        + "2. " + COMMAND_WORD + " l/bedroom\n";
+
+    private final Predicate<Item> predicate;
 
+    public ClearCommand(LocationContainsPredicate predicate) {
+        this.predicate = predicate;
+    }
+
+    public ClearCommand() {
+        this.predicate = PREDICATE_SHOW_ALL_ITEMS;
+    }
 
     @Override
-    public CommandResult execute(Model model) {
+    public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        model.setStoreMando(new StoreMando());
+        model.updateFilteredItemList(predicate);
```

Comment:
Personally, I think this does not follow the single layer of abstraction principle. Perhaps, one suggestion would be to create a method (ie; clearLocation) in the class responsible for managing the lists and call the method here.